.travis.yml

```diff
@@ -4,3 +4,19 @@ node_js:
   - '0.10'
   - '0.12'
   - 'iojs'
+
+cache:
+  directories:
+    - node_modules
+
+matrix:
+  fast_finish: true
```

Comment:
http://blog.travis-ci.com/2013-11-27-fast-finishing-builds/

> Travis CI will mark your build as finished as soon as one of two conditions are met: The only remaining jobs are allowed to fail, **or a job has already failed**. 

Yeah we don't have any allowed failures thus if a environment failed the general build will always failed.